pkg/agent/flowexporter/connections/conntrack_linux.go

```diff
@@ -66,12 +67,27 @@ func (ct *connTrackSystem) DumpFlows(zoneFilter uint16) ([]*flowexporter.Connect
 	conns, err := ct.connTrack.DumpFilter(conntrack.Filter{})
 	if err != nil {
 		klog.Errorf("Error when dumping flows from conntrack: %v", err)
-		return nil, err
+		return nil, nil, err
+	}
+
+	getMaxConnectionsInConnTrack := func() (int, error) {
+		maxConns, err := sysctl.GetSysctlNet("nf_conntrack_max")
+		if err != nil {
+			return 0, fmt.Errorf("error when getting nf_conntrack_max: %v", err)
+		}
+		return maxConns, nil
 	}
+
+	maxConns, err := getMaxConnectionsInConnTrack()
+	if err != nil {
+		klog.Errorf("Error when getting max connections in conntrack: %v", err)
```

Comment:
Yes, returning error is good enough as we log the error when there in connectionStore.Run with call to connectionStore.Poll. Please just return the error here and at L69, and do not log it. Is this what you meant @antoninbas ?

In addition, I feel one message is enough like below rather than two messages "error when getting max connections through nf_conntrack_max" when logging.

Reply:
Yes that's what I meant. You can just return the error here, it will be logged by the caller

Reply:
Got it, thanks. I find simillar situations at L60, L69 and L64, L70 in conntrack_ovs.go, shall I remove all these log messages and return the error instead?

Reply:
Yes: if you find yourself logging and returning the same error, you probably want to remove the log (after checking that someone up in the call chain will log the error). In some rare cases I guess you may want an extra log if it provides additional information.